### PR TITLE
build facets and filters

### PR DESCRIPTION
**Pull request by SimonLab**
# Ready for review

see  #111

### CODE REVIEW

**Comment by nelsonic** [comment]
moar-tests

**Comment by SimonLab** [comment]
@nelsonic do you mind to review this PR, thanks?

**Comment by nelsonic** [comment]
@SimonLab I will take a second look tomorrow morning. (_please excuse delay..._)

File: lib/transforms/search-results-to-jsonapi.js [hunk 1]
```diff
@@ -260,6 +260,137 @@ function createCountCategories (results) {
   return result;
 }
 
+/**
+* Create the facets object depending of the type of hte search (all, people, objects, documents)
+*/
+function createFacets (type, results) {
+  if (type === 'people') {
```
Comment (nelsonic) [inline lib/transforms/search-results-to-jsonapi.js]: consider using a [`switch`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/switch) ...?

also, given the format of the facet method you are invoking you could require an Object and invoke using a variable ... here's an example:

`facets.js`:
```js
module.exports = {
  'createDocumentFacets': (results) => {
    // your function here...
  },
  'createObjectFacets': (results) => {
    // your function here...
  } // ... etc.
}
```
Then when you `require` or `import` the `facets.js` you can invoke the method:
```js
const facets = require('./facets.js');
function createFacets (type, results) {
  return type ? facets['create' + type + 'Facets'](results) : {};
}
```

**Comment by nelsonic** [comment]
Code looks good. left a couple of comments, but happy to merge. 👍

**Pull request merged**